stacker/config/__init__.py

```diff
@@ -137,19 +141,12 @@ def parse(raw_config):
                     tmp_list.append(tmp_dict)
                 config_dict[top_level_key] = tmp_list
 
-    # We have to enable non-strict mode, because people may be including top
-    # level keys for re-use with stacks (e.g. including something like
-    # `common_variables: &common_variables`).
-    #
-    # The unfortunate side effect of this is that it propagates down to every
-    # schematics model, and there doesn't seem to be a good way to only disable
-    # strict mode on a single model.
-    #
-    # If we enabled strict mode, it would break backwards compatibility, so we
-    # should consider enabling this in the future.
-    strict = False
-
-    return Config(config_dict, strict=strict)
+    # Top-level excess keys are removed by Config._convert, so enabling strict
+    # mode is fine here.
+    try:
+        return Config(config_dict, strict=True)
+    except SchematicsError as e:
+        raise exceptions.InvalidConfig(e.errors)
```

Comment:
I don't know the schematics Exceptions well enough - but will this provide enough data to figure out what the issue is?  I really wish python2 provided a good way for wrapping exceptions.

Reply:
Yes, the only issue is the message is not very explanatory. `e.errors` is something like this when an invalid key is present: `{"stacks": {"0": {"garbage": "Rogue field"}}}`. But I think the best place to fix it is in Schematics itself.